### ISSUE TITLE
How to enable odometry using EKF2 estimator in v1.8.0 ?

### ISSUE THREAD

**Issue by gajena** [PX4/Firmware#9750]
I am sending odometry on `/mavros/odometry/odom` and not getting any changes in local_pose.

**Comment by gajena** [comment]
I am having one more problem when I am using EKF2 aid_mask as vision position suddenly estimator getting stop.
Should I use this branch of ecl lib `pr-extVisInit`?

**Comment by mhkabir** [comment]
Yes.

**Comment by gajena** [comment]
Now estimator is initializing but when I started to publish on `vision_pose` it's coming on `local_pose`
but after sometime `local_pose` x and y becoming zero and z become constant at last value.

**Comment by mhkabir** [comment]
The estimator is probably rejecting your data after some time. Please open a new issue with logs and details about your vision system.

**Comment by RomanBapst** [comment]
@gajena You should not be using any branches in ecl. It should just work with the release or latest master. As Kabir said, a log file would enable us to help you.

**Comment by mhkabir** [comment]
@RomanBapst Vision fusion works fine on ecl master/release, however the branch I referred to improves initialization and fixes the case where the visual estimator resets in flight. See https://github.com/PX4/ecl/pull/464.

**Comment by gajena** [comment]
@RomanBapst @mhkabir [Logs]( https://review.px4.io/plot_app?log=5d0f3bbc-a6a4-4c43-901d-ceadff5999d2#Nav-Local-Position-X )
I am done this test using ROVIO and I am not moving quad just moving VIO sensor.

**Comment by mhkabir** [comment]
> Yes, of course. The EK

**Comment by mhkabir** [comment]
Yes, they need to be a rigid body and be moved together, otherwise the EKF will reject data.

**Comment by gajena** [comment]
Okay, I will test this.
And to give commanded position should I use this topic in mavros `mavros/setpoint_position/local` ?

**Comment by tiralonghipol** [comment]
@gajena  HIi ! Any update on this ?
Thank you
